Add blog post about year 2024 (#984)
packit/packit.dev: posts/packit-in-2024/index.md

---
title: Packit in 2024
date: 2024-12-23
authors: flachman
tags:
  - 2024
  - yearly-features
  - summary
---

# Packit in 2024

We are reaching the end of the year and it’s a good time to look back at the Packit highlights of this year. And it was a year!

> We didn’t take a winter sleep, and at the very start of the year, we [announced](https://packit.dev/posts/specfile-introduction) the [Python library for parsing specfiles](https://github.com/packit/specfile). (Mainly written by [Nilola](https://github.com/nforro).) We had been successfully using it for some time and it was time to give the benefit to others. Thanks to that, there is now a [language Server for RPM spec files](https://github.com/dcermak/rpm-spec-language-server).

Reviewer: ```suggestion
We didn’t take a winter sleep, and at the very start of the year, we [announced](https://packit.dev/posts/specfile-introduction) the [Python library for parsing specfiles](https://github.com/packit/specfile). (Mainly written by [Nikola](https://github.com/nforro).) We had been successfully using it for some time and it was time to give the benefit to others. Thanks to that, there is now a [language Server for RPM spec files](https://github.com/dcermak/rpm-spec-language-server).
```

> A lot usually happens in Brussels at the turn of January and February. And this time, we were present as well for the first time. [František](https://mastodonczech.cz/@lachmanfrantisek) (joined by [Evgeny](https://github.com/ukulekek)) presented on [CentOS Connect how Packit and Testing Farm can be beneficial to the CentOS community](https://www.youtube.com/watch?v=Eqb6IqG7Jek). Also, at this conference, there was a meetup of the [CentOS Integration SIG](https://sigs.centos.org/integration/) which [Matěj](https://fosstodon.org/@m4tt_314) and [František](https://mastodonczech.cz/@lachmanfrantisek) joined. The main event in Brussels is the [FOSDEM conference](https://fosdem.org/2024/), a big conference consisting of multiple so-called rooms. [František](https://mastodonczech.cz/@lachmanfrantisek) took [an interactive session](https://fosdem.org/2024/schedule/event/fosdem-2024-2881-upstream-and-downstream-best-friends-forever-/) in the Distribution room about the relationship between upstream and downstream.

Reviewer 2: ```suggestion
A lot usually happens in Brussels at the turn of January and February. And this time, we were present as well for the first time. [František](https://mastodonczech.cz/@lachmanfrantisek) (joined by [Evgeny](https://github.com/ukulekek)) presented on [CentOS Connect how Packit and Testing Farm can be beneficial to the CentOS community](https://www.youtube.com/watch?v=Eqb6IqG7Jek). Also, at this conference, there was a meetup of the [CentOS Integration SIG](https://sigs.centos.org/integration/) which [Matej](https://fosstodon.org/@m4tt_314) and [František](https://mastodonczech.cz/@lachmanfrantisek) joined. The main event in Brussels is the [FOSDEM conference](https://fosdem.org/2024/), a big conference consisting of multiple so-called rooms. [František](https://mastodonczech.cz/@lachmanfrantisek) took [an interactive session](https://fosdem.org/2024/schedule/event/fosdem-2024-2881-upstream-and-downstream-best-friends-forever-/) in the Distribution room about the relationship between upstream and downstream.
```

Author: @mfocko do you prefer `e` or `ě`?..;)

Reviewer 3: I've decided not to die on that hill :smile:

> During the whole year, we have made many improvements so more packages can use Packit’s Fedora release automation. Since many packages can benefit from Packit right away, we’ve introduced an [onboarding](https://packit.dev/docs/cli/dist-git/init) command capable of preparing a configuration file and also moving it to dist-git – either via pull requests or direct push.

Reviewer 2: ```suggestion
During the whole year, we have made many improvements so more packages can use Packit’s Fedora release automation. Since many packages can benefit from Packit right away, we’ve introduced an [onboarding command](https://packit.dev/docs/cli/dist-git/init) capable of preparing a configuration file and also moving it to dist-git – either via pull requests or direct push.
```
probably my fault from gdoc suggestion 😅

The second half of the year was spent finishing all the initiatives we’d worked on for some time. The long-awaited feature was not to require a git upstream for Fedora release automation. This wasn’t so simple because of the old decision to build Packit on the concept of related git repositories – one representing upstream and one downstream. (And maybe one for the source-git, but that’s a [story](https://packit.dev/source-git/status) for another time.) Luckily, all the hard work was done and many packages can now use Packit.

Another feature and another change of Packit’s internal plumbings is the way to [avoid having divergent branching](https://packit.dev/posts/non-git-upstreams-support) when releasing to multiple Fedora (or EPEL) branches. Thanks to this, the commits between branches can now be shared and the history can be linear.

This is still not all. Another huge feature spreading across multiple packages is [the support for side-tags](https://packit.dev/docs/fedora-releases-guide/releasing-multiple-packages). It’s a vast improvement and enabler for various package groups that need to control many packages together. And it’s finally there!

> And this is still not all for this year. We love collaborating on features and one such collaboration was/is with [Siteshwar Vashisht](https://fosstodon.org/@siteshwar) on integrating Packit with [OpenScanHub](https://openscanhub.dev/) so Packit users can run static analysis without much effort. Later during the year, we managed to properly report differential scans (so you can check only new findings.) If you are interested in this (or would like to know why one needs to care), look at Situ’s presentation on [DevConf.CZ](https://www.youtube.com/watch?v=rcuIvAttWgY&pp=ygUdZGV2Y29uZiBzaXRlc2h3YXIgb3BlbnNjYW5odWI%3D) or [Cauldron](https://www.youtube.com/watch?v=lp7UECNg7qY) conference. You can also visit Brussels next year and [František](https://mastodonczech.cz/@lachmanfrantisek) and [Siteshwar Vashisht](https://fosstodon.org/@siteshwar) should present the integration on the [CentOS Connect](https://connect.centos.org/).

Reviewer 4: It may be worth mentioning https://github.com/packit/packit/discussions/2371 for providing feedback and discussions.

Author: Good point. Added.

> And this is still not all for this year. We love collaborating on features and one such collaboration was/is with [Siteshwar Vashisht](https://fosstodon.org/@siteshwar) on integrating Packit with [OpenScanHub](https://openscanhub.dev/) so Packit users can run static analysis without much effort. Later during the year, we managed to properly report differential scans (so you can check only new findings.) If you are interested in this (or would like to know why one needs to care), look at Situ’s presentation on [DevConf.CZ](https://www.youtube.com/watch?v=rcuIvAttWgY&pp=ygUdZGV2Y29uZiBzaXRlc2h3YXIgb3BlbnNjYW5odWI%3D) or [Cauldron](https://www.youtube.com/watch?v=lp7UECNg7qY) conference. You can also visit Brussels next year and [František](https://mastodonczech.cz/@lachmanfrantisek) and [Siteshwar Vashisht](https://fosstodon.org/@siteshwar) should present the integration on the [CentOS Connect](https://connect.centos.org/).

Reviewer 4: For the CentOS Connect talk, you may temporarily use the [review link](https://cfp.fedoraproject.org/centos-connect-2025/talk/review/EW7PWKP8E3FJPVYHUTARFZRNURYZBTDA) for our talk until the full schedule is available. Although, I understand it is not meant to be used for a wider audience.

Author: Added

> After all these big achievements and manysmall and big improvements, the [time has come to release Packit v1.0.0](https://packit.dev/posts/packit_1_0_0_action_required). Everything is now prepared, and we are giving people time to update their configuration files. (Since we love automation and pull request-based workflows, the affected users should have received a pull request with the changes required.)

Reviewer: ```suggestion
After all these big achievements and many small and big improvements, the [time has come to release Packit v1.0.0](https://packit.dev/posts/packit_1_0_0_action_required). Everything is now prepared, and we are giving people time to update their configuration files. (Since we love automation and pull request-based workflows, the affected users should have received a pull request with the changes required.)
```
